### DISCUSSION TITLE
Deploy WebAPI and WebAssembly on a Windows Server 2022 environment

### DISCUSSION THREAD

**Discussion by tvprasad** [fullstackhero/blazor-wasm-boilerplate#283] [Q&A]
Hello, 
I'm currently working on making my Blazor WebAssembly web app accessible on an corporate intranet. I'm attempting to follow the recommended hosting options, but I'm uncertain about the correct approach for hosting. I'm also planning to conduct a penetration test later to identify any security vulnerabilities.

As I delve into various articles on the subject, drawing from my experience with IIS and Apache Tomcat servers, I find myself a bit unclear about running on Kestrel and configuring a reverse proxy to expose it to users. While running apps on Kestrel, I understand there is a need for additional processes to ensure these apps persist through system reboots. Alternatively, running directly on IIS is another option, allowing IIS to handle the process.
I think community members have already running their applications in a production environment and are willing to share their experiences.
I would like to seek your guidance on the recommended way of hosting WebAPI and Blazor WebAssembly on Windows Server 2022. Your insights would be greatly appreciated.

cc: @fretje, @mluepkes , @matthewcorven, @MikaelHild

**Comment by punkouter26** [comment]
I ran it on iis at work.. I don't remember anything special.. I used the copy files publish option and put the latest .net SDKs on the iis server

**Reply by tvprasad** [reply]
@punkouter24 Thanks. Have you observed any performance issues?

**Reply by punkouter26** [reply]
no. I also published to azure as an app service just to practice..

**Reply by OnlyOneBDS** [reply]
Which project do you publish, Client or Host? I published Host thinking it would be the same as you would do for the WebAPI  project but when I did that, I get errors stating that certain files couldn't be loaded.
